dealiiadapter.cc

```diff
@@ -841,30 +918,92 @@ void ElasticProblem<dim>::output_results(const unsigned int timestep) const
     MappingQEulerian<dim> q_mapping(parameters.poly_degree, dof_handler, displacement);
     data_out.build_patches(q_mapping, parameters.poly_degree);
 
-    std::ofstream output("solution-" + std::to_string(timestep) + ".vtk");
+    std::ofstream output("dealii_output/solution-" + std::to_string(timestep) + ".vtk");
     data_out.write_vtk(output);
     std::cout<< "\t Output written to solution-" + std::to_string(timestep) + ".vtk \n" <<std::endl;
 }
 
 template <int dim>
-void ElasticProblem<dim>::initialize_precice()
+void ElasticProblem<dim>::compute_timesteps()
 {
 
-    // get precice specific IDs from precice
-    precice.configure(parameters.config_file);
+    if(parameters.enable_precice == true)
+    {
+        while(precice.isCouplingOngoing() &&
+              time.get_timestep() < time.get_n_timesteps())
+        {
+
+            time.increment();
+
+            std::cout << "  Time = " <<time.current()
+                      << " at timestep " << time.get_timestep()
+                      << " of " <<time.get_n_timesteps()
+                      << std::endl;
+
+            save_old_state();
+
+            assemble_rhs();
+
+            solve();
 
-    // Assert matching dimensions between deal.ii and precice
-    // Only valid for the current adapter setup
-    // TODO: Adapt for quasi-2D cases (#5)
-    Assert(dim == precice.getDimensions(),
-           ExcDimensionMismatch(dim, precice.getDimensions()));
+            update_displacement();
 
-    node_mesh_id = precice.getMeshID(parameters.node_mesh);
-    face_mesh_id = precice.getMeshID(parameters.face_mesh);
-    forces_data_id  = precice.getDataID(parameters.read_data_name, face_mesh_id);
-    displacements_data_id = precice.getDataID(parameters.write_data_name, node_mesh_id);
+            advance_precice();
 
+            reload_old_state();
 
+            if(precice.isTimestepComplete()
+                    && time.get_timestep() % parameters.output_interval == 0)
+                output_results(time.get_timestep());
+
+        }
+
+        precice.finalize();
+    }
+    else
+        while(time.get_timestep() < time.get_n_timesteps())
+        {
+
+            time.increment();
+
+            std::cout << "  Time = " <<time.current()
+                      << " at timestep " << time.get_timestep()
+                      << " of " <<time.get_n_timesteps()
+                      << std::endl;
+
+            assemble_rhs();
+
+            solve();
+
+            update_displacement();
+
+            if( time.get_timestep() % parameters.output_interval == 0)
+                output_results(time.get_timestep());
+        }
+}
+
+template <int dim>
+void ElasticProblem<dim>::initialize_precice()
+{
+
+    if(parameters.enable_precice == true)
```

Comment:
This is inconsistent. For advance you do the enable check outside of the function. Here, you do it in the function. Both are OK, but consistency is a must :)

Reply:
I assume for the purpose of this implementation, it is ok like this.

Reply:
As we get rid of `enable_precice`, this is no longer valid.